Rest/Content/ContentType.php

```diff
@@ -0,0 +1,379 @@
+        } elseif (!empty($options['siteAccess'])) {
+            $siteaccesses = array($options['siteAccess']);
+        } else {
+            $siteaccesses = array($this->siteAccess->name);
+        }
+
+        $subtreeCriteria = [];
+        $rootLocations = $this->siteAccessHelper->getRootLocationsBySiteaccesses($siteaccesses);
+        foreach ($rootLocations as $rootLocationId) {
+            $subtreeCriteria[] = new Criterion\Subtree($this->locationService->loadLocation($rootLocationId)->pathString);
+        }
+
+        return new Criterion\LogicalOr($subtreeCriteria);
+    }
+
+    /**
+     * Returns languages based on mandatorId or siteaccess.
+     *
+     * @param array $options
+     *
+     * @return array
+     *
+     * @throws LogicException When languages cannot be fetched from siteAccess or mandatorId.
+     */
+    private function getLanguages(array $options)
+    {
+        if (!empty($options['mandatorId'])) {
+            $languages = $this->siteAccessHelper->getMainLanguagesBySiteaccesses(
+                $this->siteAccessHelper->getSiteaccessesByMandatorId($options['mandatorId'])
+            );
+        } elseif (!empty($options['siteAccess'])) {
+            $languages = $this->configResolver->getParameter('languages', '', $options['siteAccess']);
+        } else {
+            $languages = $this->configResolver->getParameter('languages');
+        }
+
+        if (empty($languages)) {
+            throw new LogicException(sprintf('No languages found using siteAccess or mandatorId'));
+        }
+
+        return $languages;
+    }
+
+    /**
+     * Generate export files.
+     *
+     * @param $languages
+     * @param $chunkDir
+     * @param array $options
+     *
+     * @return array
+     */
+    private function generateFiles($languages, $chunkDir, array $options)
+    {
+        $urls = array();
+
+        foreach ($options['contentTypeIds'] as $contentTypeId) {
+            $count = $this->countContentByContentTypeId($contentTypeId, $options);
+
+            $this->logger->info(sprintf('fetching %s items of contentTypeId %s', $count, $contentTypeId));
+
+            foreach ($languages as $lang) {
+                $options['lang'] = $lang;
+
+                for ($i = 1; $i <= ceil($count / $options['pageSize']); ++$i) {
+                    $filename = sprintf('%d_%s_%d', $contentTypeId, $lang, $i);
+                    $options['page'] = $i;
+
+                    $content = $this->prepareContent(
+                        $this->getContentItems($options, $contentTypeId),
+                        $options
+                    );
+
+                    $chunkPath = $options['documentRoot'] . '/var/export' . $chunkDir . $filename;
+
+                    $this->generateFile($content, $chunkPath, $options);
+
+                    $url = sprintf(
+                        '%s/api/ezp/v2/ez_recommendation/v1/exportDownload%s%s',
```

Comment:
Not directly related to this PR, but this could actually be `/api/ezrecommendation/v1/exportDownload%s%s`.